src/test/java/org/mozilla/taskcluster/APITest.java

```diff
@@ -141,56 +146,57 @@ public void testDefineTask() {
-        );
-        System.out.println("=> unnamed credentials");
-        System.out.println("accessToken: "+ cred.accessToken);
-        System.out.println("clientId: "+ cred.clientId);
-        System.out.println("certificate.signature: "+cred.certificate.signature);
-        System.out.println("certificate.seed: "+cred.certificate.seed);
-        System.out.println("certificate.start: "+cred.certificate.start);
-        System.out.println("certificate.expiry: "+cred.certificate.expiry);
-        System.out.print("scopes: ");
-        for(String scope: cred.certificate.scopes){
-          System.out.print(scope + " ");
-        }
-
-        cred = Credentials.createTemporaryCredentials(
-        "clientId","issuer","tokenABCDEFGH", tempScopes, start, expiry
-        );
-
-        System.out.println("=> named credentials");
-        System.out.println("accessToken: "+ cred.accessToken);
-        System.out.println("clientId: "+ cred.clientId);
-        System.out.println("certificate.clientId: "+cred.certificate.clientId);
-        System.out.println("certificate.issuer: "+cred.certificate.issuer);
-        System.out.println("certificate.signature: "+cred.certificate.signature);
-        System.out.println("certificate.seed: "+cred.certificate.seed);
-        System.out.println("certificate.start: "+cred.certificate.start);
-        System.out.println("certificate.expiry: "+cred.certificate.expiry);
-        System.out.print("scopes: ");
-        for(String scope: cred.certificate.scopes){
-          System.out.print(scope + " ");
+    public void createTemporaryCredentials() {
+        BufferedReader br = null;
+        try {
+            Gson gson = new Gson();
+            br = new BufferedReader(new InputStreamReader(getClass().getResourceAsStream("/testcases.json")));
+            TempCredsTestCase[] testCases = gson.fromJson(br, TempCredsTestCase[].class);
+
+            for (TempCredsTestCase tc : testCases) {
+
+                log.info("Testing " + tc.description);
+                Date start = sdf.parse(tc.start);
+                Date expiry = sdf.parse(tc.expiry);
+
+                Credentials permCreds = new Credentials(tc.permCreds.clientId, tc.permCreds.accessToken, null);
+                Credentials tempCreds = permCreds.createTemporaryCredentials(tc.tempCredsName, tc.tempCredsScopes,
+                        start, expiry);
+                Certificate cert = tempCreds.getCertificate();
+
+                cert.seed = tc.seed;
+                // need to generate access token using fixed seed
+                tempCreds.accessToken = Credentials.generateTemporaryAccessToken(permCreds.accessToken, cert.seed);
+                // need to recalculate signature, as seed was updated
+                cert.generateSignature(permCreds.accessToken, tempCreds.clientId);
+                // update credentials with updated certificate
+                tempCreds.certificate = cert.toString();
```

Comment:
Created [bug 1257520](https://bugzilla.mozilla.org/show_bug.cgi?id=1257520) for adding tests against the auth endpoints.